test/metabase/test/data/sql.clj

```diff
@@ -272,9 +271,17 @@
   :hierarchy #'driver/hierarchy)
 
 (defmethod add-fk-sql :sql/test-extensions
-  [driver {:keys [database-name]} {:keys [table-name]} {dest-table-name :fk, field-name :field-name}]
-  (let [quot            #(sql.u/quote-name driver %1 (ddl.i/format-name driver %2))
-        dest-table-name (name dest-table-name)]
+  [driver {:keys [database-name] :as dbdef} {:keys [table-name]} {dest-table-name :fk, field-name :field-name}]
+  (let [quot               #(sql.u/quote-name driver %1 (ddl.i/format-name driver %2))
+        dest-table-name    (name dest-table-name)
+        dest-table-pk-name (->> dbdef
+                                :table-definitions
+                                (filter #(= (:table-name %) dest-table-name))
+                                first
```

Comment:
A `(get-table table-name table-definitions)` method would be nice to trim the density here.

Reply:
good point, created a `get-tabledef` here [98b3216](https://github.com/metabase/metabase/pull/39038/commits/98b3216404e50c36c56ccb53a61f6255b39245fe)